extendr-api/tests/deserializer_tests.rs

```diff
@@ -0,0 +1,174 @@
+#[cfg(feature = "serde")]
+mod test {
+    use extendr_api::deserializer::from_robj;
+    use extendr_api::prelude::*;
+    use serde::Deserialize;
+
+    ////////////////////////////////////////////////////////////////////////////////
+    ///
+    /// Deserialize from a Robj.
+    ///
+    /// Like JSON, we can use a Robj as a storage format.
+    ///
+    /// For example if creating vectors from a RDS file or returning a structure
+    /// or just doing a conversion.
+    ///
+    #[test]
+    fn test_deserialize_robj() {
+        test! {
+            // In these tests, the wrapper is transparent and just tests the contents.
+            // So Int(i32) is actually testing i32.
+
+            #[derive(Deserialize, PartialEq, Debug)]
+            struct Null;
+            assert_eq!(from_robj::<Null>(&r!(NULL)), Ok(Null));
+            assert_eq!(from_robj::<Null>(&r!(1)), Err(Error::ExpectedNull(r!(1))));
+
+            #[derive(Deserialize, PartialEq, Debug)]
+            struct Int(i32);
+            assert_eq!(from_robj::<Int>(&r!(1)), Ok(Int(1)));
+            assert_eq!(from_robj::<Int>(&r!(1.0)), Ok(Int(1)));
+            assert_eq!(from_robj::<Int>(&r!(NULL)).is_err(), true);
```

Comment:
Is there a reason for using `assert_eq!` when testing functions returning `bool`? Isn't it easier to use `assert!(expr)`?